src/corehost/cli/hostpolicy.cpp

```diff
@@ -20,16 +20,17 @@ int run(const arguments_t& args)
     // Load the deps resolver
     deps_resolver_t resolver(g_init, args);
 
-    if (!resolver.valid())
+    pal::string_t resolver_errors;
+    if (!resolver.valid(&resolver_errors))
     {
-        trace::error(_X("Invalid .deps file"));
+        trace::error(_X("Error initializing the dependency resolver: %s"), resolver_errors.c_str());
         return StatusCode::ResolverInitFailure;
     }
 
     pal::string_t clr_path = resolver.resolve_coreclr_dir();
     if (clr_path.empty() || !pal::realpath(&clr_path))
     {
-        trace::error(_X("Could not resolve coreclr path"));
+        trace::error(_X("Could not resolve coreclr path, turn on logging (set or export COREHOST_TRACE=1) to look for 'CoreCLR' probed paths."));
```

Comment:
`Unable to resolve CoreCLR path. For more details, enable tracing by setting the COREHOST_TRACE environment variable to '1'` (users will know how to set environment variables on their platform).

If all calls to `trace::error` result in the termination of the process, perhaps it should just call `exit` itself, and maybe have it emit the line about `COREHOST_TRACE`? For example

```
trace::error(_X("Blah blah something went wrong"), StatusCode::CoreClrResolveFailure)
```

would emit the below message and call `exit(StatusCode::CoreClrResolveFailure)` (if the `exit` refactoring would be complicated, we don't need to do it, just throwing out ideas ;)). I do think that in almost all error cases, the user is going to want to know how to get more details, so we may as well tell them about `COREHOST_TRACE` right there.

```
Blah blah something went wrong
An error occurred initializing the application. For more details, enable tracing by setting the COREHOST_TRACE environment variable to '1'`
```